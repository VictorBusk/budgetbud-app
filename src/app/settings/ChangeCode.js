import React, {Component} from 'react';
import {Keyboard, TouchableWithoutFeedback} from "react-native";
import {connect} from 'react-redux';
import {AppHeader, Separator} from "../../components";
import {
    Container,
    Text,
    Spinner,
    Button,
} from 'native-base';
import I18n from "../../strings/i18n";
import {
    button,
    text,
    container,
    color
} from "../../style";
import {showWarningToast} from "../../helpers";
import {ChangeCodeForm} from "../../components";
import {
    changeCode,
    codeChanged,
    repeatedCodeChanged,
    resetAuthState
} from "../../redux/actions";

class ChangeCode extends Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.authError) {
            showWarningToast(nextProps.authError);
            this.props.resetAuthState();
        }
    }

    onCodeChange = (text) => {
        this.props.codeChanged(text);
    };

    onRepeatedCodeChange = (text) => {
        this.props.repeatedCodeChanged(text);
    };

    onSavePress = () => {
        this.props.changeCode(
            this.props.code,
            this.props.repeatedCode, () => {
                this.props.navigation.pop();
            }
        );
    };

    render() {
        return (
            <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                <Container style={container.signedInContainer}>
                    <AppHeader headerText={I18n.t('changeCodeHeader')}
                               infoText={I18n.t('changeCodeInfo')}
                               showBackButton={true}
                               onLeftButtonPress={() => {
                                   if (!this.props.changeLoading) {
                                       this.props.navigation.pop()
                                   }
                               }}/>

                    <ChangeCodeForm
                        code={this.props.code}
                        repeatedCode={this.props.repeatedCode}
                        onCodeChange={this.onCodeChange}
                        onRepeatedCodeChange={this.onRepeatedCodeChange}
                    />

                    <Separator/>

                    <Button rounded
                            onPress={() => {
                                if (!this.props.changeLoading || !this.props.authLoading) {
                                    this.onSavePress()
                                }
                            }}
                            style={[button.defaultButton, color.button]}
                    >
                        {this.props.changeLoading ? (
                            <Spinner color='#D0D0D0'/>) : (
                            <Text style={text.submitButtonText}>
                                {I18n.t('changeCodeSaveButton')}
                            </Text>
                        )}
                    </Button>
                </Container>
            </TouchableWithoutFeedback>
        );
    }
}

const mapStateToProps = ({auth}) => {
    return {
        code,
        repeatedCode,
        authError,
        changeLoading
    } = auth;
};

const mapDispatchToProps = {
    codeChanged,
    repeatedCodeChanged,
    changeCode,
    resetAuthState
};

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCode);
