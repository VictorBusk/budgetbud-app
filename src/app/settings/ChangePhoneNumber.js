import React, {PureComponent} from 'react';
import {Keyboard, TouchableWithoutFeedback} from "react-native";
import {connect} from 'react-redux';
import {AppHeader, Separator} from "../../components";
import {
    Container,
    Text,
    Spinner,
    Button,
    Item,
    Label,
    Input,
} from 'native-base';
import I18n from "../../strings/i18n";
import {renderInputIcon} from "../../helpers";
import {
    button,
    text,
    container,
    color,
    input
} from "../../style";
import {showWarningToast} from "../../helpers";
import {
    changePhoneNumber,
    getPhoneNumber,
    phoneNumberChanged,
    resetAuthError
} from "../../redux/actions";

class ChangePhoneNumber extends PureComponent {
    componentWillMount() {
        if (!this.props.phoneNumberInitialized)
            this.props.getPhoneNumber();
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.authError) {
            showWarningToast(nextProps.authError);
            this.props.resetAuthError();
        }
    }

    onPhoneNumberChange = (text) => {
        this.props.phoneNumberChanged(text);
    };

    onSavePress = () => {
        this.props.changePhoneNumber(
            this.props.phoneNumber, () => {
                this.props.navigation.pop();
            }
        );
    };

    render() {
        return (
            <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                <Container style={container.signedInContainer}>
                    <AppHeader headerText={I18n.t('changePhoneNumberHeader')}
                               infoText={I18n.t('changePhoneNumberInfo')}
                               showBackButton={true}
                               onLeftButtonPress={() => {
                                   if (!this.props.changeLoading) {
                                       this.props.navigation.pop()
                                   }
                               }}/>

                    <Container style={{flex: 4, justifyContent: 'center'}}>
                        {this.props.authLoading ? (
                            <Spinner style={{
                                alignItems: 'center',
                                justifyContent: 'center'
                            }} color='#1c313a'/>) : (
                            <Container style={[container.defaultFormStyle, {paddingTop: 10}]}>
                                <Label style={[text.defaultText, color.text]}>{I18n.t('changePhoneNumberLabel')}</Label>
                                <Item rounded style={[input.inputField, color.input]}>
                                    <Input
                                        value={String(this.props.phoneNumber)}
                                        onChangeText={this.onPhoneNumberChange}
                                        keyboardType="numeric"
                                        maxLength={8}
                                        placeholder={I18n.t('changePhoneNumberPlaceholder')}
                                        placeholderTextColor='#7F9BAA'
                                        style={color.text}
                                    />
                                    {renderInputIcon(this.props.phoneNumber, 8)}
                                </Item>
                            </Container>)}
                    </Container>

                    <Separator/>

                    <Button rounded
                            onPress={() => {
                                if (!this.props.changeLoading || !this.props.authLoading) {
                                    this.onSavePress()
                                }
                            }}
                            style={[button.defaultButton, color.button]}
                    >
                        {this.props.changeLoading ? (
                            <Spinner color='#D0D0D0'/>) : (
                            <Text style={text.submitButtonText}>
                                {I18n.t('changePhoneNumberSaveButton')}
                            </Text>
                        )}
                    </Button>
                </Container>
            </TouchableWithoutFeedback>
        );
    }
}

const mapStateToProps = ({auth}) => {
    return {
        phoneNumber,
        authError,
        authLoading,
        changeLoading,
        phoneNumberInitialized
    } = auth;
};

const mapDispatchToProps = {
    phoneNumberChanged,
    getPhoneNumber,
    changePhoneNumber,
    resetAuthError
};

export default connect(mapStateToProps, mapDispatchToProps)(ChangePhoneNumber);
