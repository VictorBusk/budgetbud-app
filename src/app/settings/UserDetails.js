import React, {Component} from 'react';
import {connect} from 'react-redux';
import {AppHeader, ConfirmDialog} from "../../components";
import {NavigationActions} from "react-navigation";
import {
    Container,
    Content,
    List,
    ListItem,
    Label,
    Body,
    Left,
    Right,
    Icon
} from "native-base";
import I18n from '../../strings/i18n';
import {color, container} from "../../style";
import {showWarningToast} from "../../helpers";
import {
    deleteUser,
    resetAuthError,
    resetAuthCode,
    getPhoneNumber
} from "../../redux/actions";
import {debounce} from "lodash";

class UserDetails extends Component {
    componentWillMount() {
        if (!this.props.phoneNumberInitialized)
            this.props.getPhoneNumber();

        this.props.resetAuthCode();
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.authError) {
            showWarningToast(nextProps.authError);
            this.props.resetAuthError();
        }
    }

    deleteUser = () => {
        this.props.deleteUser(() => {
            this.props.navigation.dispatch(NavigationActions.reset({
                index: 0,
                key: null,
                actions: [NavigationActions.navigate({
                    routeName: "AuthStack"
                })],
            }));
        });
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <ConfirmDialog
                    title={I18n.t('confirmDialogDeletionHeader')}
                    text={I18n.t('userDetailsConfirmDialogBody')}
                    confirmCallback={() => this.deleteUser()}
                    loading={this.props.deleteUserLoading}
                    ref={(confirmDialog) => {
                        this.confirmDialog = confirmDialog
                    }}
                />

                <Container>
                    <AppHeader headerText={I18n.t('userDetailsHeader')}
                               infoText={I18n.t('userDetailsInfo')}
                               showBackButton={true}
                               onLeftButtonPress={() => this.props.navigation.pop()}/>

                    <Content>
                        <List>
                            <ListItem icon onPress={debounce(() => {
                                this.props.navigation.navigate("ChangePhoneNumber")
                            }, 400)}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-phone-portrait"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('userDetailsChangePhoneNumber')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={debounce(() => {
                                this.props.navigation.navigate("ChangeCode")
                            }, 400)}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-lock"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('userDetailsChangeCode')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={() => this.confirmDialog.showDialog()}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-trash"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('userDetailsDeleteUser')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                        </List>
                    </Content>
                </Container>
            </Container>
        );
    }
}

const mapStateToProps = ({auth}) => {
    return {
        phoneNumberInitialized,
        deleteUserLoading,
        authError
    } = auth;
};

const mapDispatchToProps = {
    deleteUser,
    resetAuthError,
    resetAuthCode,
    getPhoneNumber
};

export default connect(mapStateToProps, mapDispatchToProps)(UserDetails);
