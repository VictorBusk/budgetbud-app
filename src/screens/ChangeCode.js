import React, {PureComponent} from 'react';
import {Keyboard, TouchableWithoutFeedback} from "react-native";
import {connect} from 'react-redux';
import {AppHeader, Separator} from "../components/";
import {
    Container,
    Text,
    Spinner,
    Button,
    Item,
    Label,
    Input,
    View
} from 'native-base';
import I18n from "../strings/i18n";
import {
    codeChanged,
    repeatedCodeChanged,
    changeCode,
    resetAuthState
} from "../actions";
import {renderInputIcon} from "../helpers/validators";
import {
    button,
    text,
    container,
    color,
    input
} from "../style";
import {showWarningToast} from "../helpers/toasts";

class ChangeCode extends PureComponent {
    componentWillReceiveProps(nextProps) {
        if (nextProps.authError) {
            showWarningToast(nextProps.authError);
            this.props.resetAuthState();
        }
    }

    onCodeChange = (text) => {
        this.props.codeChanged(text);
    };

    onRepeatedCodeChange = (text) => {
        this.props.repeatedCodeChanged(text);
    };

    onSavePress = () => {
        this.props.changeCode(
            this.props.code,
            this.props.repeatedCode, () => {
                this.props.navigation.pop();
            }
        );
    };

    render() {
        return (
            <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                <Container style={container.signedInContainer}>
                    <AppHeader headerText={I18n.t('changeCodeHeader')}
                               showBackButton={true}
                               onLeftButtonPress={() => {
                                   if (!this.props.changeLoading) {
                                       this.props.navigation.pop()
                                   }
                               }}/>

                    <Container style={{flex: 4, justifyContent: 'flex-start'}}>
                        <View style={[container.defaultFormStyle, {paddingTop: 10}]}>
                            <Label style={[text.defaultText, color.text]}>{I18n.t('changeCodeLabel')}</Label>
                            <Item rounded style={[input.inputField, color.input]}>
                                <Input
                                    secureTextEntry={true}
                                    value={this.props.code}
                                    onChangeText={this.onCodeChange}
                                    keyboardType="numeric"
                                    maxLength={4}
                                    placeholder={I18n.t('changeCodePlaceholder')}
                                    placeholderTextColor='#7F9BAA'
                                    style={color.text}
                                />
                                {renderInputIcon(this.props.code, 4)}
                            </Item>
                        </View>

                        <View style={[container.defaultFormStyle, {paddingTop: 10}]}>
                            <Label style={[text.defaultText, color.text]}>{I18n.t('changeRepeatedCodeLabel')}</Label>
                            <Item rounded style={[input.inputField, color.input]}>
                                <Input
                                    secureTextEntry={true}
                                    value={this.props.repeatedCode}
                                    onChangeText={this.onRepeatedCodeChange}
                                    keyboardType="numeric"
                                    maxLength={4}
                                    placeholder={I18n.t('changeCodePlaceholder')}
                                    placeholderTextColor='#7F9BAA'
                                    style={color.text}
                                />
                                {renderInputIcon(this.props.repeatedCode, 4)}
                            </Item>
                        </View>
                    </Container>

                    <Separator/>

                    <Button rounded
                            onPress={() => {
                                if (!this.props.changeLoading || !this.props.authLoading) {
                                    this.onSavePress()
                                }
                            }}
                            style={[button.defaultButton, color.button]}
                    >
                        {this.props.changeLoading ? (
                            <Spinner color='#D0D0D0'/>) : (
                            <Text style={text.submitButtonText}>
                                {I18n.t('changeCodeSaveButton')}
                            </Text>
                        )}
                    </Button>
                </Container>
            </TouchableWithoutFeedback>
        );
    }
}

const mapStateToProps = ({auth}) => {
    return {
        code,
        repeatedCode,
        authError,
        changeLoading
    } = auth;
};

const mapDispatchToProps = {
    codeChanged,
    repeatedCodeChanged,
    changeCode,
    resetAuthState
};

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCode);
