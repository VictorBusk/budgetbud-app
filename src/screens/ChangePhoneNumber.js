import React, {PureComponent} from 'react';
import {FlatList} from 'react-native';
import {connect} from 'react-redux';
import {
    AppHeader,
    Separator
} from "../components/";
import {
    Container,
    ListItem,
    Body,
    Right,
    Text,
    Spinner,
    CheckBox,
    Button
} from 'native-base';
import I18n from "../strings/i18n";
import {
    getAccounts,
    linkAccounts,
    accountsSelected
} from "../actions";
import {
    button,
    text,
    list,
    container, color
} from "../style";

class ChangePhoneNumber extends PureComponent {
    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('changePhoneNumberHeader')}
                           showBackButton={true}
                           onLeftButtonPress={() => this.props.navigation.pop()}/>

                <Container style={{flex: 4, justifyContent: 'center'}}>
                </Container>

                <Separator/>

                <Button rounded
                        onPress={() => {
                            if (!this.props.authLoading) {
                                this.onSavePress()
                            }
                        }}
                        style={[button.defaultButton, color.button]}
                >
                    {this.props.authLoading ? (
                        <Spinner color='#D0D0D0'/>) : (
                        <Text style={text.submitButtonText}>{I18n.t('changePhoneNumberSaveButton')}</Text>
                    )}
                </Button>
            </Container>
        );
    }

    renderItem = ({item}) => {
        const checked = '';

        return (
            <ListItem>
                <Body>
                <Text style={color.text}>{item.name}</Text>
                </Body>
                <Right style={list.listItemCheckBoxPadding}>
                    <CheckBox
                        style={checked ? color.checkboxChecked : color.checkboxUnchecked}
                        checked={checked}
                        onPress={() => this.onCheckBoxPress(item)}
                    />
                </Right>
            </ListItem>
        );
    }
}

const mapStateToProps = ({auth}) => {
    const {authLoading} = auth;

    return {
        authLoading
    };
};

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(ChangePhoneNumber);
