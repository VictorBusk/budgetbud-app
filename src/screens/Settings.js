import React, {Component} from 'react';
import {
    Container,
    Content,
    List,
    ListItem,
    Label,
    Body,
    Left,
    Right,
    Icon
} from "native-base";
import {AppHeader, ConfirmDialog} from "../components/";
import {connect} from "react-redux";
import I18n from "../strings/i18n";
import {deleteBudget, resetBudgetError} from "../actions";
import {container, color} from "../style";
import {showWarningToast} from "../helpers";

class Settings extends Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.budgetError) {
            showWarningToast(nextProps.budgetError);
            this.props.resetBudgetError();
        }
    }

    deleteBudget = () => {
        const {budgetID} = this.props;
        this.props.deleteBudget({budgetID}, () => {
            this.props.navigation.navigate('CreateBudget');
        });
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <ConfirmDialog
                    title={I18n.t('confirmDialogDeletionHeader')}
                    text={I18n.t('settingsConfirmDialogBody')}
                    confirmCallback={() => this.deleteBudget()}
                    budgetLoading={this.props.budgetLoading}
                    ref={(confirmDialog) => {
                        this.confirmDialog = confirmDialog
                    }}
                />

                <Container>
                    <AppHeader headerText={I18n.t('settingsHeader')}
                               onLeftButtonPress={() => this.props.navigation.navigate("DrawerOpen")}/>

                    <Content>
                        <List>
                            <ListItem icon onPress={() => this.props.navigation.navigate("UserDetails")}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-person"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('settingsUserDetails')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={() => this.props.navigation.navigate("Accounts")}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-card"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('settingsAccounts')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={() => this.props.navigation.navigate("Alarms")}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-notifications-outline"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('settingsAlarms')}</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={() => this.confirmDialog.showDialog()}>
                                <Left>
                                    <Icon style={color.darkIcon} name="md-trash"/>
                                </Left>
                                <Body>
                                <Label style={color.text}>{I18n.t('settingsDeleteBudget')}</Label>
                                </Body>
                                <Right >
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                        </List>
                    </Content>
                </Container>
            </Container>
        );
    }
}

const mapStateToProps = ({budget}) => {
    return {budgetID, budgetError} = budget;
};

const mapDispatchToProps = {
    deleteBudget, resetBudgetError
};

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
