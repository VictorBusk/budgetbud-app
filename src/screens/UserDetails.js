import React, {Component} from 'react';
import {Container, Content, List, ListItem, Label, Body, Left, Right, Icon} from "native-base";
import {NavigationActions} from "react-navigation";
import {connect} from 'react-redux';
import {AppHeader, ConfirmDialog} from "../components/";
import {deleteUser} from "../actions";

class UserDetails extends Component {
    deleteUser = () => {
        this.props.deleteUser(() => {
            this.props.navigation.dispatch(NavigationActions.reset({
                index: 0,
                key: null,
                actions: [NavigationActions.navigate({
                    routeName: "SignIn"})],
            }));
        });
    };

    render() {
        return (
            <Container>
                <ConfirmDialog
                    title="Bekræft sletning"
                    text="Vil du slette din bruger?"
                    confirmCallback={() => this.deleteUser()}
                    loading={this.props.loading}
                    ref={(confirmDialog) => {
                    this.confirmDialog = confirmDialog }}
                />

                <Container>
                    <AppHeader headerText={'Brugeroplysninger'}
                               showBackButton={true}
                               onLeftButtonPress={() => this.props.navigation.pop()}/>

                    <Content>
                        <List>
                            <ListItem icon>
                                <Left>
                                    <Icon name="md-phone-portrait"/>
                                </Left>
                                <Body>
                                <Label>Ændr telefonnummer</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon>
                                <Left>
                                    <Icon name="md-lock"/>
                                </Left>
                                <Body>
                                <Label>Ændr pinkode</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                            <ListItem icon onPress={() => this.confirmDialog.showDialog()}>
                                <Left>
                                    <Icon name="md-trash"/>
                                </Left>
                                <Body>
                                <Label>Slet bruger</Label>
                                </Body>
                                <Right>
                                    <Icon name="arrow-forward"/>
                                </Right>
                            </ListItem>
                        </List>
                    </Content>
                </Container>
            </Container>
        );
    }
}

const mapStateToProps = ({auth}) => {
    return {loading} = auth;
};

export default connect(mapStateToProps, {
    deleteUser
})(UserDetails);
