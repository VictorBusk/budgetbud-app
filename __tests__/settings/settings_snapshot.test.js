import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChangeCode from "../../src/app/settings/ChangeCode";
import ChangePhoneNumber from "../../src/app/settings/ChangePhoneNumber";
import Settings from "../../src/app/settings/Settings";
import UserDetails from "../../src/app/settings/UserDetails";
import thunk from 'redux-thunk';
import {
    INITIAL_ACCOUNT_STATE,
    INITIAL_ALARM_STATE,
    INITIAL_AUTH_STATE,
    INITIAL_BUDGET_STATE
} from "../test_helper/initial_state";

Enzyme.configure({adapter: new Adapter()});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Settings', () => {
    it('renders ChangeCode as expected', () => {
        const wrapper = shallow(
            <ChangeCode/>,
            {context: {store: mockStore({auth: INITIAL_AUTH_STATE})}},
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders ChangeCode as expected', () => {
        const wrapper = shallow(
            <ChangePhoneNumber/>,
            {context: {store: mockStore({auth: INITIAL_AUTH_STATE})}},
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders Settings as expected', () => {
        const wrapper = shallow(
            <Settings/>,
            {
                context: {
                    store: mockStore({
                        account: INITIAL_ACCOUNT_STATE,
                        alarm: INITIAL_ALARM_STATE,
                        budget: INITIAL_BUDGET_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders UserDetails as expected', () => {
        const wrapper = shallow(
            <UserDetails/>,
            {context: {store: mockStore({auth: INITIAL_AUTH_STATE})}},
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});