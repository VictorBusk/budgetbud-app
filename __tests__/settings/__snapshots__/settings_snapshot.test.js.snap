// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders ChangeCode as expected 1`] = `
<TouchableWithoutFeedback
  accessible={false}
  onPress={[Function]}
>
  <Styled(Container)
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <AppHeader
      headerText="Change code"
      infoText="Here you can edit your code. The entered codes must be identical. Press [Save] to save the changes."
      onLeftButtonPress={[Function]}
      showBackButton={true}
    />
    <ChangeCodeForm
      code=""
      onCodeChange={[Function]}
      onRepeatedCodeChange={[Function]}
      repeatedCode=""
    />
    <Separator />
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginTop": 20,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#00263A",
          },
        ]
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "600",
          }
        }
      >
        Save
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Container)>
</TouchableWithoutFeedback>
`;

exports[`Settings renders ChangePhoneNumber as expected 1`] = `
<TouchableWithoutFeedback
  accessible={false}
  onPress={[Function]}
>
  <Styled(Container)
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <AppHeader
      headerText="Change phone number"
      infoText="Here you can edit your phone number. Press [Save] to save the changes."
      onLeftButtonPress={[Function]}
      showBackButton={true}
    />
    <Styled(Container)
      style={
        Object {
          "flex": 4,
          "justifyContent": "center",
        }
      }
    >
      <Styled(Container)
        style={
          Array [
            Object {
              "alignSelf": "center",
              "marginBottom": 5,
              "marginTop": 10,
              "width": "90%",
            },
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <Styled(Label)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 14,
                "fontWeight": "400",
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          Enter phone number
        </Styled(Label)>
        <Styled(Item)
          rounded={true}
          style={
            Array [
              Object {
                "alignSelf": "center",
                "height": 40,
                "marginVertical": 5,
                "paddingVertical": 0,
              },
              Object {
                "backgroundColor": "#F6F7F9",
                "borderColor": "#003755",
              },
            ]
          }
        >
          <Styled(Input)
            keyboardType="numeric"
            maxLength={8}
            onChangeText={[Function]}
            placeholder="12 34 56 78"
            placeholderTextColor="#7F9BAA"
            style={
              Object {
                "color": "#003755",
              }
            }
            value=""
          />
        </Styled(Item)>
      </Styled(Container)>
    </Styled(Container)>
    <Separator />
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginTop": 20,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#00263A",
          },
        ]
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "600",
          }
        }
      >
        Save
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Container)>
</TouchableWithoutFeedback>
`;

exports[`Settings renders ChangePhoneNumber as expected 2`] = `
<TouchableWithoutFeedback
  accessible={false}
  onPress={[Function]}
>
  <Styled(Container)
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <AppHeader
      headerText="Change phone number"
      infoText="Here you can edit your phone number. Press [Save] to save the changes."
      onLeftButtonPress={[Function]}
      showBackButton={true}
    />
    <Styled(Container)
      style={
        Object {
          "flex": 4,
          "justifyContent": "center",
        }
      }
    >
      <Styled(Spinner)
        color="#1c313a"
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </Styled(Container)>
    <Separator />
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginTop": 20,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#00263A",
          },
        ]
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "600",
          }
        }
      >
        Save
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Container)>
</TouchableWithoutFeedback>
`;

exports[`Settings renders Settings as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <ConfirmDialog
    confirmCallback={[Function]}
    loading={false}
    text="Do you want to delete your budget?"
    title="Confirm deletion"
  />
  <Styled(Container)>
    <AppHeader
      headerText="Settings"
      infoText="Here you can navigate to user details, accounts, and alarm settings, or delete your budget by pressing the trash can icon."
      onLeftButtonPress={[Function]}
    />
    <Styled(Content)>
      <Styled(List)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-person"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              User details
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-card"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Accounts
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-notifications-outline"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Alarms
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-trash"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Delete budget
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(List)>
    </Styled(Content)>
  </Styled(Container)>
</Styled(Container)>
`;

exports[`Settings renders UserDetails as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <ConfirmDialog
    confirmCallback={[Function]}
    loading={false}
    text="Do you want to delete your user?"
    title="Confirm deletion"
  />
  <Styled(Container)>
    <AppHeader
      headerText="User details"
      infoText="Here you can navigate to edit screens for the phone number and code. Press the trash can icon to delete your user."
      onLeftButtonPress={[Function]}
      showBackButton={true}
    />
    <Styled(Content)>
      <Styled(List)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-phone-portrait"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Change phone number
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-lock"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Change code
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)
          icon={true}
          onPress={[Function]}
        >
          <Styled(Left)>
            <Styled(Icon)
              name="md-trash"
              style={
                Object {
                  "color": "#00263A",
                }
              }
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Label)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Delete user
            </Styled(Label)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Icon)
              name="arrow-forward"
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(List)>
    </Styled(Content)>
  </Styled(Container)>
</Styled(Container)>
`;
